app/src/main/java/com/example/myapplication/HomePage.java

```diff
@@ -59,6 +71,79 @@ public void onCreate(Bundle savedInstanceState) {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_homepage, container, false);
+        binding = FragmentHomepageBinding.inflate(inflater, container, false);
+        v = container;
+        return binding.getRoot();
+    }
+    @Override
+    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        /*binding.button2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                NavHostFragment.findNavController(HomePage.this)
+                        .navigate(R.id.action_HomePage_to_pitFragment);
+            }
+        });*/
+        ObjectAnimator animation = ObjectAnimator.ofFloat(binding.WolfLogoInfo, "rotation", new float[]{0f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f});
+        animation.setDuration(1000);
+        binding.WolfLogoInfo.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                animation.start();
+                MainActivity.darkMode = !MainActivity.darkMode;
+                lightDark(v, MainActivity.darkMode);
+            }
+        });
+        lightDark(v, MainActivity.darkMode);
+    }
+    public void lightDark (ViewGroup v, boolean mode){
+        if (!mode){
```

Comment:
Some notes about the lightDark method for improvement:
- your if/else statment is almost an exact copy except for small value changes
- and it's written multiple times each in a different class, so this is something that can be simplified by putting it into a helper class and simplifying the if/else like such:

`public class UIHelpers {
    public static void lightDark (ViewGroup v, boolean mode){
        int bgColor = (mode ? (Color.BLACK) : Color.WHITE);
        int textColor = (mode ? (Color.WHITE) : (Color.BLACK));
        for (int i = 0; i < v.getChildCount(); i ++){
            View child = v.getChildAt(i);
            if (!(child instanceof Button)) {
                child.setBackgroundColor(bgColor);
                if (child instanceof TextView) {
                    TextView tx = (TextView) child;
                    tx.setTextColor(textColor);
                }
            }
            if (child instanceof Switch) {
                Switch tx = (Switch) child;
                tx.setTextColor(textColor);
            }
            if (child instanceof ViewGroup){
                lightDark((ViewGroup)child, mode);
            }
        }
    }
}
`